Tribler/Core/Modules/restapi/display_endpoint.py

```diff
@@ -77,10 +103,23 @@ def render_GET(self, request):
         :return: the node data formatted in JSON
         """
         if "focus_node" not in request.args:
-            return DisplayEndpoint.return_400(request, "focus_node parameter missing")
+            return DisplayEndpoint.return_error(request, message="focus_node parameter missing")
 
         if len(request.args["focus_node"]) < 1 or len(request.args["focus_node"][0]) == 0:
-            return DisplayEndpoint.return_400(request, "focus_node parameter empty")
+            return DisplayEndpoint.return_error(request, message="focus_node parameter empty")
+
+        focus_node = request.args["focus_node"][0]
+        if isinstance(focus_node, basestring) and not focus_node.lstrip("-").isdigit():
```

Comment:
Although input is validated in roughly the same way as written here, other endpoints do delegate this to deeper in the program. I'll leave this check out for now and act the same as other endpoints.